catalog/spatial/kml/spatial-kml-transformer/src/main/java/org/codice/ddf/spatial/kml/transformer/KMLTransformerImpl.java

```diff
@@ -208,6 +228,89 @@ protected Placemark performDefaultTransformation(Metacard entry)
     return kmlPlacemark;
   }
 
+  private void setExtendedData(Placemark placemark, Metacard metacard) {
+    final ExtendedData extendedData = new ExtendedData();
+
+    final Set<AttributeDescriptor> attributeDescriptors =
+        metacard.getMetacardType().getAttributeDescriptors();
+
+    for (AttributeDescriptor attributeDescriptor : attributeDescriptors) {
+      final String attributeName = attributeDescriptor.getName();
+      final Attribute attribute = metacard.getAttribute(attributeName);
+      if (attribute != null) {
+        Serializable convertedAttribute = convertAttribute(attribute, attributeDescriptor);
+        if (convertedAttribute == null) {
+          LOGGER.debug("Attribute {} converted to null value.", attributeName);
+        } else {
+          String attributeValue = convertedAttribute.toString();
+          attributeValue = StringUtils.removeStart(StringUtils.removeEnd(attributeValue, "]"), "[");
```

Comment:
It's possible for an attribute (e.g. one with a String value) to begin or end with these characters. Since `convertAttribute` can return either a list or a single value, these could be mistakenly removed. If you performed the String conversion in `convertAttribute` you could strip these out only when the attribute is multivalued.